pkg/datapath/loader/netlink.go

```diff
@@ -151,3 +155,164 @@ func RemoveTCFilters(ifName string, tcDir uint32) error {
+	}
+
+	sysSettings := make([]sysctl.Setting, 0, 5)
+	if option.Config.EnableIPv6 {
+		sysSettings = append(sysSettings, sysctl.Setting{
+			Name: fmt.Sprintf("net.ipv6.conf.%s.forwarding", ifName), Val: "1", IgnoreErr: false})
+	}
+	if option.Config.EnableIPv4 {
+		sysSettings = append(sysSettings, []sysctl.Setting{
+			{Name: fmt.Sprintf("net.ipv4.conf.%s.forwarding", ifName), Val: "1", IgnoreErr: false},
+			{Name: fmt.Sprintf("net.ipv4.conf.%s.rp_filter", ifName), Val: "0", IgnoreErr: false},
+			{Name: fmt.Sprintf("net.ipv4.conf.%s.accept_local", ifName), Val: "1", IgnoreErr: false},
+			{Name: fmt.Sprintf("net.ipv4.conf.%s.send_redirects", ifName), Val: "0", IgnoreErr: false},
+		}...)
+	}
+	if err := sysctl.ApplySettings(sysSettings); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func setupDevs(links ...netlink.Link) error {
+	for _, link := range links {
+		if err := setupDev(link); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func setupVethPair(name, peerName string) error {
+	// Create the veth pair if it doesn't exist.
+	if _, err := netlink.LinkByName(name); err != nil {
+		hostMac, err := mac.GenerateRandMAC()
+		if err != nil {
+			return err
+		}
+		peerMac, err := mac.GenerateRandMAC()
+		if err != nil {
+			return err
+		}
+
+		veth := &netlink.Veth{
+			LinkAttrs: netlink.LinkAttrs{
+				Name:         name,
+				HardwareAddr: net.HardwareAddr(hostMac),
+			},
+			PeerName:         peerName,
+			PeerHardwareAddr: net.HardwareAddr(peerMac),
+		}
+		if err := netlink.LinkAdd(veth); err != nil {
+			return err
+		}
+	}
+
+	veth, err := netlink.LinkByName(name)
+	if err != nil {
+		return err
+	}
+	if err := setupDev(veth); err != nil {
+		return err
+	}
+	peer, err := netlink.LinkByName(peerName)
+	if err != nil {
+		return err
+	}
+	if err := setupDev(peer); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func setupIpvlan(name string, nativeLink netlink.Link) (*netlink.IPVlan, error) {
+	hostLink, err := netlink.LinkByName(name)
+	if err == nil {
+		// Ignore the error.
+		netlink.LinkDel(hostLink)
+	}
+
+	ipvlan := &netlink.IPVlan{
+		LinkAttrs: netlink.LinkAttrs{
+			Name:        name,
+			ParentIndex: nativeLink.Attrs().Index,
+		},
+		Mode: netlink.IPVLAN_MODE_L3,
+	}
+	if err := netlink.LinkAdd(ipvlan); err != nil {
+		return nil, err
+	}
+
+	if err := setupDev(ipvlan); err != nil {
+		return nil, err
+	}
+
+	return ipvlan, nil
+}
+
+// setupBaseDevice decides which and what kind of interfaces should be set up as
+// the first step of datapath initialization, then performs the setup (and
+// creation, if needed) of those interfaces. It returns two links and an error.
+// By default, it sets up the veth pair - cilium_host and cilium_net.
+// In flannel mode, it sets up the native interface used by flannel.
+// In ipvlan mode, it creates the cilium_host ipvlan with the native device as a
+// parent.
+func setupBaseDevice(nativeDevs []netlink.Link, mode string, mtu int) (netlink.Link, netlink.Link, error) {
```

Comment:
I would suggest to replace the `mode string` with `mode baseDeviceMode` where `baseDeviceMode` is `type baseDeviceMode string`